abpframework/abp
# CLI: Check latest stable version from a new endpoint

### Description

Instead of retrieving the latest stable version from NuGet,  from now on we will get it from a new endpoint. After we release a new stable version, we should update the `latest-version.json` file in the root directory of this repository and when a user runs the CLI it will check from that file.

This will give us time to test new versions before notifying the developers and allow us to take necessary actions if we faced any problems.

> I haven't found any other places that make latest stable version check, if you know any other places that we should update, please let me know. @ebicoglu @yekalkan 

### Checklist

- [X] ~I documented it (or no need to document or I will create a separate documentation issue)~ - **no need to document**

### How to test it?

You can check the following steps for testing this PR:

* Remove the **!DEBUG** pre-processor from https://github.com/abpframework/abp/blob/dev/framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/CliService.cs#L58.
* Change [this URL](https://github.com/abpframework/abp/blob/88c6f0b1ddf55df37426e3f81bafb0483aba3b57/framework/src/Volo.Abp.Cli.Core/Volo/Abp/Cli/CliUrls.cs#L11) to "**https://raw.githubusercontent.com/abpframework/abp/1e21e0f756f60fda1865b2323403323641f85400/latest-versions.json**".

> You can lower your ABP CLI version such as *6.0.0* and run the `Volo.Abp.Cli` project, after following the steps above.
